go/vt/vtctl/grpcvtctldserver/server_test.go

```diff
@@ -8811,3 +8813,633 @@ func TestValidateShard(t *testing.T) {
+				Sharded: true,
+				Tables: map[string]*vschemapb.Table{
+					"t1": {
+						Columns: []*vschemapb.Column{
+							{
+								Name: "c1",
+								Type: querypb.Type_BINARY,
+							},
+						},
+						ColumnVindexes: []*vschemapb.ColumnVindex{
+							{
+								Column: "c1",
+								Name:   "c1",
+							},
+						},
+					},
+				},
+				Vindexes: map[string]*vschemapb.Vindex{
+					"c1": {
+						Type: "hash",
+					},
+				},
+			},
+			includeVSchema: false,
+			shouldErr:      false,
+		},
+		{
+			name: "different schemas",
+			schemas: []*tabletmanagerdatapb.SchemaDefinition{
+				{
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name:    "t1",
+							Columns: []string{"c1"},
+						},
+					},
+				},
+				{
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name:    "t2",
+							Columns: []string{"c1"},
+						},
+					},
+				},
+				{
+					TableDefinitions: []*tabletmanagerdatapb.TableDefinition{
+						{
+							Name:    "t3",
+							Columns: []string{"c1"},
+						},
+					},
+				},
+			},
+			vschema: &vschemapb.Keyspace{
+				Sharded: true,
+				Tables: map[string]*vschemapb.Table{
+					"t1": {
+						Columns: []*vschemapb.Column{
+							{
+								Name: "c1",
+								Type: querypb.Type_BINARY,
+							},
+						},
+						ColumnVindexes: []*vschemapb.ColumnVindex{
+							{
+								Column: "c1",
+								Name:   "c1",
+							},
+						},
+					},
+				},
+				Vindexes: map[string]*vschemapb.Vindex{
+					"c1": {
+						Type: "hash",
+					},
+				},
+			},
+			includeVSchema: false,
+			shouldErr:      true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Test Setup
+			ts, _ := memorytopo.NewServerAndFactory("zone1")
+			tmc := &testutil.TabletManagerClient{
+				TopoServer: ts,
+				GetSchemaResults: map[string]struct {
+					Schema *tabletmanagerdatapb.SchemaDefinition
+					Error  error
+				}{},
+			}
+			vtctld := testutil.NewVtctldServerWithTabletManagerClient(t, ts, tmc, func(ts *topo.Server) vtctlservicepb.VtctldServer {
+				return NewVtctldServer(ts)
+			})
+
+			// Create Keyspace
+			req := &vtctldatapb.CreateKeyspaceRequest{
+				Name: "ks",
+				Type: topodatapb.KeyspaceType_NORMAL,
+			}
+
+			_, err := vtctld.CreateKeyspace(ctx, req)
+			require.NoError(t, err, "error creating keyspace 'ks'")
+
+			// Create Shard
+			req2 := &vtctldatapb.CreateShardRequest{
+				Keyspace:  "ks",
+				ShardName: "-80",
+			}
+			_, err = vtctld.CreateShard(ctx, req2)
+			require.NoError(t, err, "error creating ks/-80")
+
+			for i, schema := range tt.schemas {
+				tablet := &topodatapb.Tablet{
+					Alias: &topodatapb.TabletAlias{
+						Cell: "zone1",
+						Uid:  uint32(1111111111 + i),
+					},
+					Keyspace:  "ks",
+					Shard:     "-80",
+					MysqlPort: 3306,
+				}
+				tmc.GetSchemaResults["zone1-"+strconv.Itoa(int(tablet.Alias.Uid))] = struct {
```

Comment:
see my previous comment about configuring per-test. don't share this state among test cases, it makes adding future cases harder to customize